### ISSUE TITLE
Implement Filtering in the UI

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar#215]
**Problem**

The user might want to filter, sort, and group data when viewing a table.

**Proposed solution**

We should implement these options [per the design spec](https://wiki.mathesar.org/design/specs/filter-sort-group).

**Additional context**

Design work is complete (see #160). 

Also blocked by backend work in
#157 (done)
#158 (done)
#216 (done)

@pavish please feel free to split this up into smaller issues.

**Comment by pavish** [comment]
This issue has been split into smaller issues.

Implement Grouping in the UI  #343 
Implement Sorting in the UI #344
